### ISSUE TITLE
smart commands act on root partition not root device

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#1206]
When gathering smart info the partition device name for root is used rather than the base device name.
This is due to the exception made for root, when installed on a partition (default), of storing the partition name rather than the base device name ie sda3 rather than sda. Rather than loose this information we could make the smart routines robust to these calls. Note that mostly these partition rather than base device calls function as expected. However in some instances when calling smartctl on a device's partition we loose information eg:
```
/usr/sbin/smartctl -l selftest -l selective /dev/sdc3
smartctl 6.2 2013-07-26 r3841 [x86_64-linux-4.3.3-1.el7.elrepo.x86_64] (local build)
Copyright (C) 2002-13, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF READ SMART DATA SECTION ===
Device does not support Self Test logging
```
and we have a return code of 4 (via $?)

Where as if we use the base device name of sdc:
```
/usr/sbin/smartctl -l selftest -l selective /dev/sdc
smartctl 6.2 2013-07-26 r3841 [x86_64-linux-4.3.3-1.el7.elrepo.x86_64] (local build)
Copyright (C) 2002-13, Bruce Allen, Christian Franke, www.smartmontools.org

/dev/sdc: Unknown USB bridge [0x0781:0x5580 (0x010)]
Please specify device type with the -d option.

Use smartctl -h to get a usage summary
```
we have a generic return code of 1 (via $?) and in the second instance we have helpful info on what is wrong and a suggestion on what to try which would be nice not to loose.

These examples were as a result of an install to a SanDisk Extreme USB 3.0 which as the second example suggests, requires custom smartctl options (-d sat as it goes); but with the current call to partition this is not surfaced. Hence the suggestion we move to calling only on base devices.

I am currently looking into this issue and hope to make some more progress tomorrow (UK).

I am considering this issue as a precursor to #1079 as if sorted then the suggestion that additional custom smart options may be required will be surfaced.

**Comment by phillxnet** [comment]
I am currently working on this issue.

**Comment by phillxnet** [comment]
Turns our this issue was also providing false positives of smart availability when no useful info was actually available without custom smart options.

**Comment by phillxnet** [comment]
A parsing issue I'm sorting while here is that of inadvertent smart value trimming when values contain a ": " in the Identity tab (from smartctrl -H --info devname):
The following two values show the result:
```
In smartctl database [for details use
SATA 3.0, 6.0 Gb/s (current
```
when they should read:
```
In smartctl database [for details use: -P show]
SATA 3.0, 6.0 Gb/s (current: 6.0 Gb/s)
```

**Comment by phillxnet** [comment]
Closing as fixed by merged pr #1212
